Plugin/Tax/Config/ShippingTaxPlugin.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FireGento\MageSetup\Plugin\Tax\Config;
+
+use FireGento\MageSetup\Model\System\Config\Source\Tax\Dynamic as FireGentoSource;
+use Magento\Customer\Model\ResourceModel\GroupRepository;
+use Magento\Customer\Model\Session;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Quote\Api\Data\ShippingAssignmentInterface;
+use Magento\Quote\Model\Quote\Item;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\Store;
+use Magento\Tax\Api\Data\TaxClassKeyInterface;
+use Magento\Tax\Model\Calculation\Proxy;
+
+/**
+ * Class ShippingTaxPlugin
+ *
+ * FireGento\MageSetup\Plugin\Tax\Config
+ */
+class ShippingTaxPlugin
+{
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var Session
+     */
+    private $customerSession;
+
+    /**
+     * @var GroupRepository
+     */
+    private $groupRepository;
+
+    /**
+     * @var Proxy
+     */
+    private $taxCalculation;
+
+    /**
+     * @var \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory
+     */
+    private $taxClassKeyDataObjectFactory;
+
+    /**
+     * Constructor class
+     *
+     * @param ScopeConfigInterface                              $scopeConfig
+     * @param Session                                           $customerSession
+     * @param GroupRepository                                   $groupRepository
+     * @param \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory $taxClassKeyDataObjectFactory
+     */
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        Session $customerSession,
+        GroupRepository $groupRepository,
+        \Magento\Tax\Api\Data\TaxClassKeyInterfaceFactory $taxClassKeyDataObjectFactory
+    ) {
+        $this->scopeConfig                  = $scopeConfig;
+        $this->customerSession              = $customerSession;
```

Comment:
Could you define a [proxy](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/proxies.html) for this?

Reply:
I'm not using customer session anymore, data is pulled from Quote or Shipping-Assignment objects.
This allows less dependencies and more consistent operation between backend and fronend.